### ISSUE TITLE
Updater: No download progress hooks

### ISSUE THREAD

**Issue by jku** [theupdateframework/tuf#1057]
**Description of feature request**:
For large downloads an interactive client would typically want to provide progress information (progress bar or just a running percentage or count of bytes). tuf.Updater could provide functionality that enables this.

The practical case I'm looking at is pip which currently uses the progress module while downloading chunks with requests module. If tuf support is added to pip, it will lose the progress indication functionality.

**Current behavior**:
The downloads are done in chunks already but there does not seem to be any way to add hooks into the download loop.

**Expected behavior**:
Updater could e.g. call an optional hook function with some progress data periodically. At a minimum the number of bytes downloaded so far should be available.

**Comment by lukpueh** [comment]
Thanks for the great feature request, @jku! Let me cross-reference with this general updater stack clean-up issue (https://github.com/theupdateframework/tuf/issues/841) and another issue that touches the networking aspects (#932).

**Comment by jku** [comment]
When/if this is fixed, the default download settings should be re-evaluated:
```
CHUNK_SIZE = 400000 #bytes
MIN_AVERAGE_DOWNLOAD_SPEED = 50 #bytes/second
SOCKET_TIMEOUT = 4 #seconds
```

A large download at 50b/sec seems to mean one progress update every two hours which would not be very interactive.

Of course the values also mean that a server sending a large file 1 b every 4 seconds would only get flagged as slow retrieval attack _after 18 days of downloading_... so maybe these values could be re-evaluated anyway. 

Joshua pointed out #932 as related

**Comment by jku** [comment]
This would be deprecated with #1213 _updater: abstract out the network IO_

**Comment by joshuagl** [comment]
Resolved obsolete by #1213
